controllers/asosecret_controller_test.go

```diff
@@ -115,6 +115,29 @@ func TestASOSecretReconcile(t *testing.T) {
 				}
 			}),
 		},
+		"should reconcile normally for AzureClusterIdentity without spec.clientSecret": {
```

Comment:
This test actually passes without the fix, presumably because the fake client doesn't return throw a `resource name may not be empty` error. It still seems like a worthwhile case to check though.